src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -56,8 +56,8 @@ internal static class DotNetSdkLocationHelper
             // in the .NET 5 SDK rely on the .NET 5.0 runtime. Assuming the runtime that shipped with a particular SDK has the same version,
             // this ensures that we don't choose an SDK that doesn't work with the runtime of the chosen application. This is not guaranteed
             // to always work but should work for now.
-            if (major > Environment.Version.Major ||
-                (major == Environment.Version.Major && minor > Environment.Version.Minor))
+            if (major > Environment.Version.Major || (major == Environment.Version.Major && minor > Environment.Version.Minor) 
+                && !allowQueryAllRuntimeVersions)
```

Comment:
I can never remember boolean operator precedence rules. How would you feel about reordering this?

```suggestion
            if (!allowQueryAllRuntimeVersions && 
                (major > Environment.Version.Major || 
                 (major == Environment.Version.Major && minor > Environment.Version.Minor)))
```